CitationStyleLanguagePlugin.inc.php

```diff
@@ -325,63 +380,135 @@ public function getArticleTemplateData($hookName, $args)
+                        $currentAuthor->given = htmlspecialchars($chapterAuthor->getLocalizedGivenName());
+                    }
 
-        $abbreviation = $context->getData('abbreviation', $context->getPrimaryLocale()) ?? $context->getData('acronym', $context->getPrimaryLocale());
-        if ($abbreviation) {
-            $citationData->{'container-title-short'} = htmlspecialchars($abbreviation);
+                    $userGroup = $chapterAuthor->getUserGroup();
+                    if ($userGroup && in_array($userGroup->getId(), $chapterAuthorGroups, false)) {
+                        $citationData->author[] = $currentAuthor;
+                    }
+                }
+                break;
+            //OMP book
+            case $this->isApplicationOmp():
+                $citationData->type = 'book';
+                $citationData->risType = 'BOOK';
+                $citationData->id = $submission->getId();
+                $citationData->title = $publication->getLocalizedFullTitle();
+                $citationData = $this->addSeriesInformation($citationData, $publication);
+                $citationData->publisher = htmlspecialchars($context->getLocalizedName());
+                $citationData->keywords = $keywords[AppLocale::getLocale()] ?? [];
+                if ($publication->getData('pages')) {
+                    $citationData->page = htmlspecialchars($publication->getData('pages'));
+                }
+                $citationData->abstract = htmlspecialchars(strip_tags($publication->getLocalizedData('abstract')));
+                $citationData->serialNumber = $this->getSerialNumber($publication);
+                break;
+            //OJS article
+            default:
+                $citationData->type = 'article-journal';
+                $citationData->risType = 'JOUR';
+                $citationData->id = $submission->getId();
+                $citationData->title = $publication->getLocalizedFullTitle();
+                $citationData->{'container-title'} = $context->getLocalizedName();
+                $issue = $issue ?? Repo::issue()->get($publication->getData('issueId'));
+                if ($issue) {
+                    $citationData->volume = htmlspecialchars($issue->getData('volume'));
+                    // Zotero prefers issue and Mendeley uses `number` to store revisions
+                    $citationData->issue = htmlspecialchars($issue->getData('number'));
+                }
+
+                $sectionDao = Application::getSectionDAO();
+                if ($sectionId = $publication->getData('sectionId')) {
+                    $section = $sectionDao->getById($sectionId);
+                    if ($section && !$section->getHideTitle()) {
+                        $citationData->section = htmlspecialchars($section->getTitle($context->getPrimaryLocale()));
+                    }
+                }
+                $citationData->keywords = $keywords[AppLocale::getLocale()] ?? [];
```

Comment:
Yes, but only for books and not for chapters. It's done in line 445. I think, for now, there are no keywords for chapters in OMP. But it could be interesting in future to add a possibility to add keywords to chapters. I think its only a matter of time till somebody ask for chapter keywords.

Reply:
Ah I see it. Thanks!